### PR TITLE
Make CennzNut open for extension

### PR DESCRIPTION
**Pull request by hoani**
Restructure in preparation for new features to be added to cennznut

- Explicitly force methods and modules to belong to a version of cennznut
- Make ValidationErr open to take multiple domains
- Migrated V0 specific tests under v0 (still some room for improvement here)

### CODE REVIEW

File: src/validation.rs [hunk 1]
```diff
@@ -17,16 +17,16 @@ extern crate std as alloc;
 use alloc::fmt::{self, Display, Formatter};
 use pact::interpreter::types::PactType;
 
-use crate::Domain;
+// use crate::Domain;
```
Comment (cowboy-bebug) [inline src/validation.rs]: This can be removed 😄

File: src/cennznut/v0/mod.rs [hunk 2]
```diff
@@ -14,19 +14,26 @@ extern crate alloc;
 #[cfg(feature = "std")]
 extern crate std as alloc;
 
+#[cfg(test)]
+pub mod tests;
```
Comment (holygits) [inline src/cennznut/v0/mod.rs]: Does this need to be `pub`?

Comment (hoani) [reply src/cennznut/v0/mod.rs]: No, changed to private

File: src/validation.rs (hunk 1 above)
Comment (hoani) [reply src/validation.rs]: Done :)

**Pull request merged**